src/Kestrel.Core/Internal/Http/IHttpParser.cs

```diff
@@ -11,5 +11,9 @@ public interface IHttpParser<TRequestHandler> where TRequestHandler : IHttpHeade
         bool ParseRequestLine(TRequestHandler handler, in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined);
 
         bool ParseHeaders(TRequestHandler handler, in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined, out int consumedBytes);
+
+        bool ParseRequestLine(TRequestHandler handler, ref BufferReader<byte> reader);
```

Comment:
Should the interface have 2 `ParseRequestLine` methods?

Reply:
I guess this is temporary to avoid breaking many existing usages of old methods.

Reply:
Its versioning the interface by adding two new methods so its breaking them anyway

Reply:
I'm unclear on the ramifications of removing the old one- outside of work. :) Are there external dependencies on this interface?

Reply:
Its in the `.Internal` namespace, so its ok to break it :)

Reply:
@benaadams I meant inside kestrel repo, tests/benchmarks/etc.

@JeremyKuhne no, removing them is fine.

Reply:
Ok, working my way through it.